ingest/indexer/service/writelistener/bank_write_listener.go

```diff
@@ -0,0 +1,82 @@
+package writelistener
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	"github.com/osmosis-labs/osmosis/osmomath"
+	"github.com/osmosis-labs/osmosis/v25/ingest/indexer/domain"
+	indexerdomain "github.com/osmosis-labs/osmosis/v25/ingest/indexer/domain"
+)
+
+var _ storetypes.WriteListener = (*bankWriteListener)(nil)
+
+type bankWriteListener struct {
+
+	// shared context to handle graceful shutdown in case of node stop.
+	ctx context.Context
+
+	client indexerdomain.PubSubClient
+
+	coldStartManager domain.ColdStartManager
+}
+
+func NewBank(ctx context.Context, client indexerdomain.PubSubClient, coldStartManager domain.ColdStartManager) storetypes.WriteListener {
+	return &bankWriteListener{
+		ctx:    ctx,
+		client: client,
+
+		coldStartManager: coldStartManager,
+	}
+}
+
+// OnWrite implements types.WriteListener.
+func (s *bankWriteListener) OnWrite(storeKey storetypes.StoreKey, key []byte, value []byte, delete bool) error {
```

Comment:
Question - since I'm not familiar with the bank module, could you please help me understand what this function is doing? Thanks.

Reply:
Remember how we discussed two cases:
1. Cold start -> ingest all supply and supply offsets
2. Ingest only the updates

This function assumes case 2. If it detects 1 already happened, it listens to state updates of supply and supply offsets kv store keys and publishes them to the pub-sub client